### DISCUSSION TITLE
Restore photo date the photo was taken

### DISCUSSION THREAD

**Discussion by micash545** [immich-app/immich#2972] [Feature Request]
### The feature

When the photo has no date when it was taken, server sets the oldest date from metadata. 

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bcsteeve** [comment]
This probably isn't that simple, as there's a fair to good chance that the data just isn't there, and the data it would have would be irrelevant (for example, the date it was put on the system or the date it was modified by the system or last accessed).

I think a good feature might be a tool that allows users to set a date though.  It seems to me that exiftool is already part of the distribution, so some user-level access to it might be amazing!  Particularly if it can be coupled with the multi-select feature that already exists.  (ie. I select bulk photos, click "set date" and have exiftool update all the pictures).

Hmm, that would allow us to geo-code photos missing that data too!  Oh man, imagine being able to use the existing map feature to select an area of the map to update the geocode data?!  That would be awesome.

**Comment by scorpio862** [comment]
I was just about to write a new feature request for manually setting the date of photo then I noticed this discussion. 

A lot of photos that I have backed up from my phone, or from my camera, has a wrong date for some reasons.

On the time line it shows with the date of for example downloading image from messaging apps or maybe with the date of exporting image from Photoshop or the date on the camera wasn't set correctly. So often the date of the photo is not exactly the date when the photo was really taken. 

It would be great if there was an option of selecting multiple photos and setting manually the date. This way the photo would show in the correct place on the time line.

**Reply by bcsteeve** [reply]
It would be a killer feature, in my opinion.  Exiftool is powerful. Not only can you set an arbitrary date, but you can "shift" dates, which is fantastic if your camera's date was off (or say you forgot to adjust when you crossed a timezone).  I recently spent DAYS doing this on my collection, and having that functionally within Immich would be a game changer.  Exiftool is NOT user friendly and is command line only.  There have been Gui projects but they appear to be obsolete.  I bet there would be people who would switch to Immich JUST for a user-friendly way to change exif info!

Potential "pro" feature?

**Comment by bo0tzz** [comment]
Closing as a duplicate of #1696
